src/placeos-core/process_check.cr

```diff
@@ -52,54 +58,71 @@ module PlaceOS::Core
               # If there's an empty response, the modules that were meant to be running are not.
               # This is taken as a sign that the process is dead.
               # Alternatively, if the response times out, the process is dead.
+              info = nil
               timeout(PROCESS_COMMS_TIMEOUT) do
-                if protocol_manager.info.empty?
-                  State::Dead
-                else
-                  State::Running
-                end
+                info = protocol_manager.info
```

Comment:
Assuming #info is getting info from some other fibre, then that other fibre will need to yield to the scheduler somehow and reasonably frequently for the timeout to actually work.